fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/NamespaceTools.java

```diff
@@ -90,6 +94,12 @@ public static void validatePath(final Session session, final String path) {
         final String relPath = path.replaceAll("^/+", "").replaceAll("/+$", "");
         final String[] pathSegments = relPath.split("/");
         for (final String segment : pathSegments) {
+            // Empty path elements are not valid!
```

Comment:
I would take it even further in saying that this test should exist deeper in the core such that it is performed regardless of whether the request is over HTTP or native Java.

I also, completely agree in principle that this check breaks the design and intent of the `IdentifierConverter`. However, I believe consistency outweighs design principle in this case given the following:
- NamespaceTools#validatePath() is called in two places: 
* https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java#L129
* https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/NodeServiceImpl.java#L57

#validatePath() called from HttpResourceConverter#doForward() is already doing something similar to what you are asking for.
#validatePath() called from NodeServiceImpl#exists() is currently used in 6 different places in the RESTful services: FedoraLdp and FedoraNodes. These calls from RESTful services to NodeServiceImpl#exists() are the source of the real issue of breaking the `IdentifierConverter` design.

I think the future refactoring of the #exists() calls to be more in line with the `IdentifierConverter` design will be much simpler and cleaner if we keep the validation logic consolidated versus scattering it throughout the codebase in the short term.

Your other comments can be trivially addressed once we get this question sorted.